### PR TITLE
Switch march to mcpu for ARM

### PR DESCRIPTION
**Pull request by ianshmean**
See https://github.com/JuliaCI/julia-buildbot/issues/148

cc. @staticfloat

### CODE REVIEW

File: master/inventory.py [hunk 1]
```diff
@@ -151,15 +154,17 @@ def build_names(platform, arch, name):
...
         nthreads = 8
         flags += 'JULIA_CPU_TARGET=generic '
 
     # Add MARCH to flags
     if not march is None:
         flags += "MARCH=%s "%(march)
-
+    # Add MCPU to flags
+    if not mcpu is None:
+        flags += "MCPU=%s "%(mcpu)
```
Comment (staticfloat) [inline master/inventory.py:167]: I don't think the Julia buildsystem uses `MCPU`, does it?  We [have `MARCH` defined](https://github.com/JuliaLang/julia/blob/master/Make.inc#L671), but there's no hint of `MCPU` anywhere.  I think you'd need to plumb it through down here to be added to the compiler invocations: https://github.com/JuliaLang/julia/blob/86ee57cdd77d402ea63108c16df7117d0ce83dc7/Make.inc#L820-L822

File: master/inventory.py [hunk 2]
```diff
@@ -126,7 +129,7 @@ def build_names(platform, arch, name):
 
     if '-armv7l-' in name:
         tar_arch = 'armv7l'
-        march = 'armv7-a'
+        mcpu = 'armv7-a'     # Better to set mcpu on arm https://github.com/JuliaCI/julia-buildbot/issues/148
```
Comment (staticfloat) [inline master/inventory.py:132]:
```suggestion
        # Better to set mcpu than march on arm https://github.com/JuliaCI/julia-buildbot/issues/148
        mcpu = 'armv7-a'
```

File: master/inventory.py (hunk 1 above)
Comment (ianshmean) [reply master/inventory.py:167]: Yeah, sorry forgot to link the PR I did here https://github.com/JuliaLang/julia/pull/35617

**Comment by ianshmean** [comment]
Corresponding changes in base to support MCPU during build https://github.com/JuliaLang/julia/pull/35617

**Pull request merged**